src/linear_solver_logs/solve_log_ma.jl

```diff
@@ -244,6 +246,32 @@ function update_ma!(log::LSLogMA, res::Union{AbstractVector, Real}, lambda_base:
             push!(log.resid_hist, accum / ma_info.lambda) 
             push!(log.iota_hist, accum2 / ma_info.lambda) 
         end
+    
+    elseif ma_info.lambda == ma_info.lambda1 && !ma_info.flag
+        diff = ma_info.idx - ma_info.lambda
+        # Determine start point for first loop
+        startp1 = diff < 0 ? 1 : (diff + 1)
+
+        # Determine start and endpoints for second loop
+        startp2 = diff < 0 ? ma_info.lambda2 + diff + 1 : 2 
+        endp2 = diff < 0 ? ma_info.lambda2 : 1 
+        # Compute the moving average two loop setup required when lambda < lambda2
+        for i in startp1:ma_info.idx
+            accum += ma_info.res_window[i]
+            accum2 += ma_info.res_window[i]^2
+        end
+
+        for i in startp2:endp2
+            accum += ma_info.res_window[i]
+            accum2 += ma_info.res_window[i]^2
+        end
+
+        #Update the log variable with the information for this update
+        if mod(iter, log.collection_rate) == 0 || iter == 0
+            push!(log.lambda_hist, ma_info.lambda)
+            push!(log.resid_hist, accum / ma_info.lambda) 
+            push!(log.iota_hist, accum2 / ma_info.lambda) 
+        end
```

Comment:
This conditional statement needs more commentary. When is this accessed?

Reply:
I see now that this is really unclear, I have added comments that I think clarify what is going on